spec.bs

```diff
@@ -385,6 +448,31 @@ report=] unless their [=aggregatable report/debug details=] differ.
 Issue: Unique internal value is not an exported definition. See
     <a href="https://github.com/whatwg/infra/issues/583">infra/583</a>.
 
+<h3 dfn-type=dfn>Budgeting scope</h3>
+A budgeting scope defines a maximum amount of budget that a site is permitted to
+consume over a duration of time. Specifically, it defines an a upper bound on
+the <math><msub><mi>L</mi><mn>1</mn></msub></math> norm of a site's
+contributions' values, per API, over a rolling time window. It is a [=struct=]
+with the following items:
+
+<dl dfn-for="budgeting scope">
+: <dfn>max budget</dfn>
+:: A positive integer
+: <dfn ignore>duration</dfn>
+:: A [=duration=]
+: <dfn ignore>caller type</dfn>
+:: A [=context type=]
+
+</dl>
+
+Issue: Currently, this spec only references [=budgeting scope/max budget=]. We
+need to decide whether deleting the unused fields will improve clarity.
+
+<h3 dfn-type=dfn>Named budget ledger</h3>
```

Comment:
I suppose "ledger" is inaccurate because it doesn't record individual "transactions". WDYT about calling it a *Named budget balance* or maybe just a *Named balance*?

Reply:
I think I might want to generalize even further. Sometimes I want these maps from strings to int to track budget remaining, but sometimes it's useful to track budget consumed. I think it might reduce the cognitive load of reading some of the new algorithm additions if we're able to say things like `[=ledger/Subtract=] |budgetConsumed| from |budgetRemaining|`. Anyway, that's still TBD.

Reply:
Ack -- I think my one concern is just that the description "A named budget ledger tracks the remaining budget per named budget" doesn't quite address that the ledger entry starts unadjusted for previous usage, and that we also need to limit based on overall usage (not just per-name). But if we clarify that, then I think counting 'down' is fine. (Another naming option: named budget allocation?)